aptos-move/block-executor/src/executor.rs

```diff
@@ -109,45 +111,52 @@ where
     }
 
     fn execute(
-        &self,
         version: Version,
         signature_verified_block: &[T],
         last_input_output: &TxnLastInputOutput<T::Key, E::Output, E::Error>,
         versioned_cache: &MVHashMap<T::Key, T::Value, X>,
         scheduler: &Scheduler,
         executor: &E,
-        base_view: &S,
+        sync_view: &mut LatestView<T, S, X>,
     ) -> SchedulerTask {
         let _timer = TASK_EXECUTE_SECONDS.start_timer();
         let (idx_to_execute, incarnation) = version;
         let txn = &signature_verified_block[idx_to_execute as usize];
 
-        let speculative_view = MVHashMapView::new(versioned_cache, scheduler);
```

Comment:
I didn't understand this. Why is `speculative_view` removed?

Reply:
instead of creating every time, I was re-using and just updating index, but reverted back now.